core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -206,42 +212,42 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
               Future.successful(WorkspaceAccessLevels.NoAccess)
             }
 
-          DBIO.from(accessLevelFuture()) flatMap { accessLevel =>
+          traceDBIOWithParent("accessLevelFuture", s2)(_ => DBIO.from(accessLevelFuture())) flatMap { accessLevel =>
             // we may have calculated accessLevel because canShare/canCompute needs it;
             // but if the user didn't ask for it, don't return it
             val optionalAccessLevelForResponse = if (options.contains("accessLevel")) { Option(accessLevel) } else { None }
 
             // determine which functions to use for the various part of the response
             def bucketOptionsFuture(): Future[Option[WorkspaceBucketOptions]] = if (options.contains("bucketOptions")) {
-              gcsDAO.getBucketDetails(workspaceContext.workspace.bucketName, RawlsBillingProjectName(workspaceContext.workspace.namespace)).map(Option(_))
+              traceWithParent("getBucketDetails",s2)(_ =>  gcsDAO.getBucketDetails(workspaceContext.workspace.bucketName, RawlsBillingProjectName(workspaceContext.workspace.namespace)).map(Option(_)))
```

Comment:
since these are inside the `accessLevelFuture`, shouldn't they use its span as their parent?

Reply:
this tracing is actually just for the `accessLevelFuture` call (the close paren is before the flatmap)

Reply:
ah I see it, thanks. So then the `s3` can go back to `_`. Thanks!